### PR TITLE
(all) fix synchronization issues with stage details rendering

### PR DESCRIPTION
**Pull request by anotherchrisberry**
When `synchronizeSections` is called, it potentially changes the route, it often calls `$state.go`, which is asynchronous, and the new stage is not available in the scope in the current digest cycle.

We've worked around this in a piecemeal manner in stage details controllers, wrapping parts of the `initialize` methods in controllers in `$timeout` calls. 

These changes, while not perfect, standardizes the stage controller structure, providing a method to invoke after the state has been updated by the `executionDetailsSectionService`.

There are a couple of pending PRs out for openstack that I'd like to get merged before I push this, since there will be conflicts and I'd rather deal with them than make other folks do so.

It's worth revisiting the way we structure stage modules to reduce a lot of the boilerplate code that's been copy/pasted everywhere on stage details and stage configuration. Just not today.

@zanthrash @icfantv @danielpeach PTAL.

### CODE REVIEW

File: app/scripts/modules/amazon/pipeline/stages/bake/bakeExecutionDetails.controller.js [hunk 1]
```diff
@@ -1,34 +1,30 @@
 'use strict';
 
+import detailsSectionModule from '../../../../core/delivery/details/executionDetailsSection.service.ts';
```
Comment (icfantv) [inline app/scripts/modules/amazon/pipeline/stages/bake/bakeExecutionDetails.controller.js]: please remove `.ts` extension

File: app/scripts/modules/amazon/pipeline/stages/cloneServerGroup/cloneServerGroupExecutionDetails.controller.js [hunk 2]
```diff
@@ -1,66 +1,65 @@
 'use strict';
 
 import _ from 'lodash';
+import detailsSectionModule from '../../../../core/delivery/details/executionDetailsSection.service.ts';
```
Comment (icfantv) [inline app/scripts/modules/amazon/pipeline/stages/cloneServerGroup/cloneServerGroupExecutionDetails.controller.js]: ditto

File: app/scripts/modules/amazon/pipeline/stages/disableCluster/disableClusterExecutionDetails.controller.js [hunk 3]
```diff
@@ -1,23 +1,26 @@
 'use strict';
 
+import detailsSectionModule from '../../../../core/delivery/details/executionDetailsSection.service.ts';
```
Comment (icfantv) [inline app/scripts/modules/amazon/pipeline/stages/disableCluster/disableClusterExecutionDetails.controller.js]: ditto

**Comment by anotherchrisberry** [comment]
I removed the `.ts` everywhere but needed to install codelyzer first - included that in a separate PR (https://github.com/spinnaker/deck/pull/2807)

**Comment by icfantv** [comment]
crap, i had that in package.json before destroying the failed branch from earlier and (clearly) missed in my manual copy/paste back over.  my bad.

**Comment by anotherchrisberry** [comment]
heh, no worries

**Comment by icfantv** [comment]
and it was still in my node_modules directory.  i'm surprised travis didn't complain...

**Comment by anotherchrisberry** [comment]
We have Travis configured to cache the `node_modules` directory between builds, so, if you had it in the `package.json` on one of previous builds of the PR, it would have gotten cached and used by subsequent builds.

That's my guess, anyway. I looked at the build logs and didn't see any errors.

**Pull request merged**